R-package/R/osf-interactions.R

```diff
@@ -0,0 +1,174 @@
+# Load utility functions
+source(file.path(getwd(), "R", "create-release-files.R"))
+
+#' Update OSF files
+#'
+#' This function updates files on the Open Science Framework (OSF) using the provided OSF link and token.
+#' It downloads the dictionary file, validates the dictionary version and files sheet, and stages the files to be uploaded to OSF.
+#'
+#' @param OSF_REPO_LINK The link to the OSF repo.
+#' @param OSF_TOKEN The OSF token used to authenticate the user.
+#' @param dictionary_path The path to the dictionary file. If NULL, the file will be downloaded from OSF.
+#' @param past_dictionary_path The path to the past dictionary file. If NULL, the file will be downloaded from OSF.
+#'
+update_osf <- function(OSF_REPO_LINK,
+                       OSF_TOKEN,
+                       dictionary_path = NULL,
+                       past_dictionary_path = NULL){
+  # Setup logging
+  # Remove previous log
+  file.remove(odm_dictionary$log_path)
+  # Set logger appending
+  logger::log_appender(logger::appender_file(odm_dictionary$log_path))
+  
+  # Download file using passed credentials
+  dictionary_path <- download_dictionary(dictionary_path, OSF_TOKEN, OSF_REPO_LINK, odm_dictionary$tmp_dictionary_directory, origin_directory = "dev-release")
```

Comment:
I think this line and the lines that parses the dictionary should be pulled into a function called `download_dev_dictionary` in an osf module. This module would be responsible for interactions with the osf repo. This function would also be used in the `create_release_files` function.